engine/oam.asm

```diff
@@ -564,8 +599,34 @@ OAMManager_Init:
     ldy #0
 
     @Loop:
+        pha
         jsr OAMObject_Init      ; Initialize the current OAM object (X register points to it)
-        sty oam_object.index, X ; Set the index of the object to its position in the array
+
+        A8
+        tya
+        sta oam_object.index, X ; Set the index of the object to its position in the array
+
+        ; Compute word offset into the OAM low table (16 bits per object)
+        tya 
```

Comment:
do you need this extra tya?

Reply:
No, just keep the code organized in terms of having blocks of "prepare accumulator, do something with accumulator" but I will remove it